packages/teams-js/src/private/filesExtensionsFor3PStorageproviders.ts

```diff
@@ -0,0 +1,194 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { decodeAttachment } from '../internal/mediaUtil';
+import { ErrorCode, SdkError } from '../public';
+import { FrameContexts } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * Extended files API 3P storage providers, features like sending Blob from Teams to 3P app on user
+ * actions like drag and drop to compose
+ * @beta
+ */
+export namespace filesExtensionsFor3PStorageproviders {
+  /**
+   * Object used to represent a file
+   * @beta
+   *
+   */
+  export interface FilesFor3PApps extends Blob {
+    /**
+     * A number that represents the number of milliseconds since the Unix epoch
+     */
+    lastModified: number;
+    /**
+     * Name of the file
+     */
+    name: string;
+    /**
+     * file type
+     */
+    type: string;
+    /**
+     * A string containing the path of the file relative to the ancestor directory the user selected
+     */
+    webkitRelativePath?: string;
+  }
+  export interface FileChunk {
+    /**
+     * Base 64 data for the requested uri
+     */
+    chunk: string;
+    /**
+     * chunk sequence number
+     */
+    chunkSequence: number;
+  }
+  export interface FileResult {
+    /**
+     * error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type which is getting recieved
+     */
+    fileType: string;
```

Comment:
Is `fileType` a mimeType? If so, can you please update the comment stating which mimeTypes are supported and which are not and provide the link to that?

Reply:
Here fileType refers to all types of files. (i.e. Word, PDF, etc)

Reply:
I can upload an `.exe` file as well? If so, won't that be a security issue? Getting something malicious files can damage the internal systems as well as external users. Can you elaborate on how this is handled?

Reply:
In our current drag and drop (when file gets uploaded in onedrive), user is able to upload .exe file.
When the user uploads .exe file, we are only storing that file in our Appollo cache and when the request is made from 3P app we will fetch the file from cache and send to teams js in chunk and from teams js it will go to 3P app.
All users cannot send the files in any 3P app. If admin has allowed his user to upload in 3P then only user will have this capability.
In current feature, BOX team wants to store their data in BOX so they will set this property for all of their users.
Since we are only sending file, so it should not cause any security issue at teams side.